examples/FMP.jl

```diff
@@ -0,0 +1,443 @@
+# # Force Based Motion Planning
+# ![](circle_swap_pretty.gif)
+
+# This is an example implementation of the [Force Based Motion Planning (FMP)](https://arxiv.org/pdf/1909.05415.pdf) algorithm. The algorithm is a decentralized motion
+# planning algorithm where individual agents experience attractive forces towards
+# a goal position and repulsive forces from other agents or objects in the environment.
+# The FMP algorithm is included as a predefined `update_vel` function for use
+# with [`ContinuousSpace`](@ref).
+
+
+# ## Defining the Agent
+
+# First we need to define an `AbstractAgent` struct that has the necessary
+# information for the FMP algorithm to run. `pos`, and `vel` are position and
+# velocity, respectively. `tau` is an agents goal position. `color` is an
+# optional parameter for plotting used in this example. `type` is the agent
+# type: `:A` is an agent, `:O` is an obstacle, and `:T` is a target. `radius`
+# is the radius of the agent. `SSdims` is an optional parameter used for
+# plotting purposes.
+#
+# Agent types have the following distinctions:
+# - :A = agent (has velocity, subject to repulsive forces)
+# - :O = object (has no velocity, cannot move, agents must go around)
+# - :T = target (no velocity, just a projection of agent target zone)
+
+using Agents, Random, LinearAlgebra
+
+mutable struct FMP_Agent <: AbstractAgent
+    id::Int
+    pos::NTuple{2, Float64}
+    vel::NTuple{2, Float64}
+    tau::NTuple{2, Float64}
+    color::String
+    type::Symbol
+    radius::Float64
+    SSdims::NTuple{2, Float64}  ## include this for plotting
+    Ni::Array{Int64} ## array of tuples with agent ids and agent positions
+    Gi::Vector{Int64} ## array of neighboring goal IDs
+end
+
+# ## Defining the Model
+
+# Next, we define the model. The FMP algorithm has several parameters defined
+# in the original paper which are stored in a struct. Typical values can be
+# generated using the `FMP_Parameter_Init` function and loaded into the model
```

Comment:
`FMP_Parameter_Init` is mentioned both here and in the source code but doesn't exist. It has to be replaced with `fmp_parameter_init`.